src/pynwb/legacy/map.py

```diff
@@ -25,54 +25,93 @@ class TypeMapLegacy(TypeMap):
 
     def get_builder_dt(self, builder):
 
-        if builder.name == 'roi_ids':
-            pass
-        elif builder.name == 'root':
+        if builder.name == 'root':
             return 'NWBFile'
-        attrs = builder.attributes
-        ndt = attrs.get('neurodata_type')
-        if ndt == 'Module':
-            return 'ProcessingModule'
-        elif ndt == 'TimeSeries':
-            ancestry = attrs['ancestry']
-            if decode(ancestry[-1]) == 'TwoPhotonSeries' and decode(builder.name) == 'corrected':
-                return 'ImageSeries'
-            else:
-                return decode(ancestry[-1])
-
-        elif ndt == 'Interface':
-            return builder.name
-        if ndt == 'Epoch':
-            return 'Epoch'
         else:
-            parent_ndt = self.get_builder_dt(builder.parent)
-            if parent_ndt == 'Epoch':
-                return 'EpochTimeSeries'
-            if parent_ndt == 'MotionCorrection':
-                return 'CorrectedImageStack'
-            if parent_ndt == 'ImagingPlane' and isinstance(builder, GroupBuilder):
-                return 'OpticalChannel'
-            if parent_ndt == 'ImageSegmentation':
-                if builder.name in ('roi_ids', 'cell_specimen_ids'):
+
+            attrs = builder.attributes
+            ndt = attrs.get('neurodata_type')
+
+            if ndt == 'Module':
+                return 'ProcessingModule'
+            elif ndt == 'Interface':
+                return builder.name
+            elif ndt == 'Epoch':
+                return 'Epoch'
+            elif ndt == 'TimeSeries':
+                ancestry = decode(attrs['ancestry'][-1])
+                if ancestry == 'TwoPhotonSeries' and decode(builder.name) == 'corrected':
+                    return 'ImageSeries'
+                else:
+                    return ancestry
+            elif ndt == 'Custom':
```

Comment:
And may be a function named `def get_builder_type_from_custom_neurodata_type` or similar name.

That said, it is also nice to have all type handling in one function, and add the `noqa` exception can be legitimate.

Reply:
same as above